### DISCUSSION TITLE
Type Reconstruction

### DISCUSSION THREAD

**Discussion by Bohun9** [ppolesiuk/type-systems-notes#79] [Q&A]
I think I found a problem in one of the rules.
![image](https://github.com/ppolesiuk/type-systems-notes/assets/79287762/368bc095-0d1a-4827-a02c-b5ca0b732feb)

Equations in $\theta_1$ should be respected during inferring of $e_2$ and currently they are not.

Example program with this issue:
$\lambda x . ((\lambda a . \text{let } y = x + 1 \text{ in } a) (\text{let } y = (\text{not } x) \text{ in } x))$

**Comment by Bohun9** [comment]
I sent pull request with a fix.

**Reply by ppolesiuk** [reply]
There was a typo in the rule. I've merged the pull request.
